src/graph/optimizer/OptimizerUtils.cpp

```diff
@@ -912,6 +917,32 @@ bool OptimizerUtils::findOptimalIndex(const Expression* condition,
   return true;
 }
 
+// Check if the relational expression has a valid index
+// The left operand should either be a kEdgeProperty or kTagProperty expr
+bool OptimizerUtils::relExprHasIndex(
+    const Expression* expr,
+    const std::vector<std::shared_ptr<nebula::meta::cpp2::IndexItem>>& indexItems) {
+  DCHECK(expr->isRelExpr());
+
+  for (auto& index : indexItems) {
+    const auto& fields = index->get_fields();
+    if (fields.empty()) {
+      return false;
+    }
+
+    auto left = static_cast<const RelationalExpression*>(expr)->left();
+    DCHECK(left->kind() == Expression::Kind::kEdgeProperty ||
+           left->kind() == Expression::Kind::kTagProperty);
+
+    auto propExpr = static_cast<const PropertyExpression*>(left);
+    if (propExpr->prop() == fields[0].get_name()) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
```

Comment:
Please add test case as below:
tag t1 (c1, c2)
tag t2(c1, c2)
index i1 on t1 (c1, c2)
index i2 on t2 (c3, c4)
lookup on t2 where c1 in (.....)  ----> this logic will return true if the guess is correct, Because index column name can be the same from different indexes.

Reply:
Now we check the schema type in LookupValidator

```
(root@nebula) [nba]> lookup on player where team.name == "Tim Duncan"
[ERROR (-1009)]: SemanticError: Schema name error: team
```